cloudcix_primitives/snapshot_lxd.py

```diff
@@ -14,18 +15,309 @@
+                
+        except Exception as e:
+            return False, fmt.store_error(str(e), f"{prefix+3}: {messages[prefix+3]}{str(e)}"), fmt.successful_payloads, result
+
+        return True, '', fmt.successful_payloads, result
+
+    status, msg, successful_payloads, result = run_host(endpoint_url, 3220, {})
+
+    if not status:
+        return False, {}, msg
+    
+    # Prepare return data
+    data = {endpoint_url: result}
+    
+    # Check if snapshot found
+    if result.get('snapshot') is None:
+        return True, data, f'1201: {messages[1201]}'
+    else:
+        return True, data, f'1200: {messages[1200]}'
+
+
+def scrub(
+        endpoint_url: str,
+        project: str,
+        container_name: str,
+        snapshot_name: str,
+        verify_lxd_certs: bool = True
+) -> Tuple[bool, str, dict]:
+    """Delete a snapshot from an LXD container.
+    
+    :param endpoint_url: The endpoint URL for the LXD Host.
+    :param project: The LXD project name.
+    :param instance_name: The name of the LXD container containing the snapshot.
+    :param snapshot_name: The name of the snapshot to delete.
+    :param verify_lxd_certs: Boolean to verify LXD certs.
+    
+    :return: A tuple with a boolean flag indicating success or failure, a message, and a dictionary of successful payloads.
+    """
+    # Define messages
+    messages = {
+        1100: f'Successfully deleted snapshot {snapshot_name} from container {container_name} on {endpoint_url}',
+        3121: f'Failed to connect to {endpoint_url} for containers.get payload',
+        3122: f'Failed to run containers.get payload on {endpoint_url}',
+        3123: f'Failed to find snapshot {snapshot_name} for container {container_name}',
+        3124: f'Failed to delete snapshot {snapshot_name}. Error: ',
+    }
+
+    def run_host(endpoint_url, prefix, successful_payloads):
+        rcc = LXDCommsWrapper(comms_lxd, endpoint_url, verify_lxd_certs, project)
+        fmt = HostErrorFormatter(
+            endpoint_url,
+            {'payload_message': 'STDOUT', 'payload_error': 'STDERR'},
+            successful_payloads,
+        )
+        
+        # Get the instance
+        ret = rcc.run(cli='containers.get', name=container_name)
+        if ret["channel_code"] != CHANNEL_SUCCESS:
+            return False, fmt.channel_error(ret, f"{prefix+1}: {messages[prefix+1]}"), fmt.successful_payloads
+        if ret["payload_code"] != API_SUCCESS:
+            return False, fmt.payload_error(ret, f"{prefix+2}: {messages[prefix+2]}"), fmt.successful_payloads
+
+        instance = ret['payload_message']
+        fmt.add_successful('containers.get', ret)
+        
+        # Find and delete the snapshot
+        try:
+            # Get the snapshot
+            snapshot = instance.snapshots.get(snapshot_name)
+            if not snapshot:
+                return False, f"{prefix+3}: {messages[prefix+3]}", fmt.successful_payloads
```

Comment:
As snapshot does not exist, this can return True as it is the expected result from a scrub request -  (idempotentency)

Refer to feedback on Closed MR - https://github.com/CloudCIX/primitives/pull/89#discussion_r1983844992